packages/world-consumer/src/experimental/WorldConsumer.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { RESOURCE_TABLE, RESOURCE_OFFCHAIN_TABLE } from "@latticexyz/store/src/storeResourceTypes.sol";
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+import { ResourceIds } from "@latticexyz/store/src/codegen/tables/ResourceIds.sol";
+import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
+import { ResourceAccess } from "@latticexyz/world/src/codegen/tables/ResourceAccess.sol";
+import { IBaseWorld } from "@latticexyz/world/src/codegen/interfaces/IBaseWorld.sol";
+import { WorldResourceIdLib } from "@latticexyz/world/src/WorldResourceId.sol";
+import { WorldContextConsumer } from "@latticexyz/world/src/WorldContext.sol";
+import { System } from "@latticexyz/world/src/System.sol";
+
+abstract contract WorldConsumer is System {
+  bytes14 public immutable namespace;
```

Comment:
do we need this anymore if we can now do `namespaceId.getNamespace()`?

Reply:
oh does immutable not have the same read gas cost?

Reply:
yeah immutable actually works similar to constants, the data gets added to the bytecode so you are not reading from storage. That reminds me that immutable vars shouldn't be readable from within the constructor, so not sure why the tests are passing 🤔

Reply:
<img width="704" alt="image" src="https://github.com/user-attachments/assets/20de6ef6-be38-4133-bdd3-7fcdb69969c7" />

TIL how `immutable` works

Reply:
Ah since solidity 0.8.21 they can be read in the constructor, all good